src/twisted/internet/interfaces.py

```diff
@@ -728,7 +728,7 @@ def lookupZone(name, timeout=None):
 
 class IReactorTCP(Interface):
 
-    def listenTCP(port, factory, backlog=50, interface=''):
+    def listenTCP(port, factory, backlog=50, interface='', listenMultiple=False):
```

Comment:
This is changing an interface to add a new optional parameter.

This is a backwards incompatible change -- since someone *external* to Twisted who has written their own reactor is now no longer complying with this interface.

If we want to do a backwards incompatible change, we should go by the appropriate process -- probably following the "exceptions" section in https://twistedmatrix.com/documents/current/core/development/policy/compatibility-policy.html#procedure-for-exceptions-to-this-policy

An alternative is to add reuse port in a separate interface (e.g., :code:`IReactorTCPListenMulti`), which we would only add to the reactor if we detected SO_REUSEPORT in socket)

Also -- presumably we would like to be able to REUSEPORT on IPv6 as well?
(Either way, whether we do a backwards incompatible change or if we add a new interface,
it would be worthwhile to chunk those into one change.)

OTOH -- I'm not that sure we need the support in listenSSL -- since that functionality can easily be accomplished with the txsni plugin (and we probably want to get away from SSL having to lock-step with TCP)